mlrun/serving/states.py

```diff
@@ -1672,7 +1731,30 @@ def preprocess_event(self, event):
 
     def select_runnables(self, event):
         models = cast(list[Model], self.runnables)
-        return self.model_selector.select(event, models)
+        return self.model_runner_selector.select_models(event, models)
+
+    def select_outlets(self, event) -> Optional[Collection[str]]:
+        sys_outlets = [f"{self.name}_error_raise"]
+        if "background_task_status_step" in self._name_to_outlet:
+            sys_outlets.append("background_task_status_step")
+        if self._raise_exception and self._is_error(event):
+            return sys_outlets
+        user_outlets = self.model_runner_selector.select_outlets(event)
+        if user_outlets:
+            return (
+                user_outlets if isinstance(user_outlets, list) else [user_outlets]
+            ) + sys_outlets
+        return None
+
+    def _is_error(self, event: dict) -> bool:
+        if len(self.runnables) == 1:
+            if isinstance(event, dict):
+                return event.get("error") is not None
+        else:
+            for model in event:
```

Comment:
Bug: The `return` statement is inside the loop, so it returns on the first model checked regardless of result.

Example: If first model has no error but second model has error → returns `False` incorrectly.

Suggested fix:
```python
else:
    for model in event:
        body_by_model = event.get(model)
        if isinstance(body_by_model, dict) and "error" in body_by_model:
            return True
    return False
```

Reply:
@alxtkr77 sorry missed this one.. will fix in follow up pr